lib/Sema/TypeCheckType.cpp

```diff
@@ -785,6 +785,36 @@ static Type diagnoseUnknownType(TypeChecker &tc, DeclContext *dc,
   }
 
   // Qualified lookup case.
+
+  // Try ignoring access control.
+  NameLookupOptions lookupOptions = defaultMemberLookupOptions;
```

Comment:
The rest of the options are slightly different. Really it probably makes more sense to pass it down from outside.